skyportal/handlers/api/instrument.py

```diff
@@ -208,3 +246,23 @@ def delete(self, instrument_id):
               application/json:
                 schema: Error
         """
+
+
+def add_instrument_tiles(
+    instrument_id, request_handler, field_data, field_of_view_shape, field_of_view_size
+):
+    try:
+        tile_args = {'instrument_id': int(instrument_id)}
+        mocs = get_mocs(field_data, field_of_view_shape, field_of_view_size)
+        fields = [
+            InstrumentField.from_moc(moc, field_id=int(field_id), tile_args=tile_args)
+            for field_id, moc in zip(field_data['ID'], mocs)
+        ]
+        for field in fields:
+            field.instrument_id = int(instrument_id)
```

Comment:
Why is this necessary? You're already passing in instrument ID in tile_args above. And can't you just pass in `instrument_id` directly in the `InstrumentField` instantiation above?

Reply:
InstrumentField.from_moc doesn't have a instrument_id input.

Reply:
Ah, okay.